### PR TITLE
Working on Sync & Send - when switching from foreground to background

### PR DESCRIPTION
**Pull request by juanky201271**
* adding a flag for sending in RPC class.
* including the permission for `SYSTEM_ALERT_WINDOW`, Just in case.
* New `BackgroundErrorType` to store a background error, and then the App can shows to the User when the App is again in foreground. This is done for all the `Alerts`. (for sending & shielding)
* adding `activityIndicator` when the App is syncing while sending. If you can see this spin component, you know the send process is working as expected.

### CODE REVIEW

**Comment by juanky201271** [comment]
this PR fixed this situation:
https://github.com/zingolabs/zingo-mobile/issues/355#issuecomment-1561820758

**Comment by zancas** [comment]
@fluidvanadium @Oscar-Pepper can y'all please prioritize this?

**Review by zancas — COMMENTED** [review]
I would really like to see an e2e test for this!

**Comment by juanky201271** [comment]
> I would really like to see an e2e test for this!

ok, I'm going to try to do this use case:
- open a funded wallet.
- wait for syncing.
- when the wallet have funds, send some.
- put the app in background.
- wait I don't know how much time to finish the `send while sync` process.
- hopefully I got an error.
- bring the App to foreground.
- check the Alert with the error.

sounds like doable????? let's see...

File: app/LoadedApp/LoadedApp.tsx [hunk 1]
```diff
@@ -883,7 +887,12 @@ class LoadedAppClass extends Component<LoadedAppClassProps, AppStateLoaded> {
     //console.log("jc change", resultStr);
     if (resultStr.toLowerCase().startsWith('error')) {
       //console.log(`Error change wallet. ${resultStr}`);
-      Alert.alert(this.props.translate('loadedapp.changingwallet-label') as string, resultStr);
+      const background = await AsyncStorage.getItem('@background');
+      if (background === 'yes') {
+        this.setBackgroundError(this.props.translate('loadedapp.changingwallet-label') as string, resultStr);
+      } else {
+        Alert.alert(this.props.translate('loadedapp.changingwallet-label') as string, resultStr);
+      }
```
Comment (fluidvanadium) [inline app/LoadedApp/LoadedApp.tsx]: why is this logic repeated? can it become a dry function?

**Comment by fluidvanadium** [comment]
just now tested this. i hit send, got some new indicators, switched to background, switched to foreground, successfully sent

File: app/LoadedApp/LoadedApp.tsx (hunk 1 above)
Comment (juanky201271) [reply app/LoadedApp/LoadedApp.tsx]: I did that, and I think is ready to review/merge.

**Comment by fluidvanadium** [comment]
tested again, followed this use case `ok, I'm going to try to do this use case:
open a funded wallet.
wait for syncing.
when the wallet have funds, send some.
put the app in background.
wait I don't know how much time to finish the send while sync process.
hopefully I got an error.
bring the App to foreground.
check the Alert with the error.`

**Comment by fluidvanadium** [comment]
> tested again, followed this use case `ok, I'm going to try to do this use case: open a funded wallet. wait for syncing. when the wallet have funds, send some. put the app in background. wait I don't know how much time to finish the send while sync process. hopefully I got an error. bring the App to foreground. check the Alert with the error.`

manual test success

**Review by fluidvanadium — APPROVED** [review]

**Review by zancas — APPROVED** [review]

**Pull request merged**